provider/cluster/kube/client.go

```diff
@@ -187,26 +199,42 @@ func (c *client) Deploy(ctx context.Context, lid mtypes.LeaseID, group *manifest
 
 	for svcIdx := range group.Services {
 		service := &group.Services[svcIdx]
-		if err := applyDeployment(ctx, c.kc, newDeploymentBuilder(c.log, c.settings, lid, group, service)); err != nil {
-			c.log.Error("applying deployment", "err", err, "lease", lid, "service", service.Name)
-			return err
+
+		persistent := false
+		for i := range service.Resources.Storage {
+			attrVal := service.Resources.Storage[i].Attributes.Find(builder.StorageAttributePersistent)
+			if persistent, _ = attrVal.AsBool(); persistent {
+				break
+			}
+		}
+
+		if persistent {
+			if err := applyStatefulSet(ctx, c.kc, builder.NewStatefulSet(c.log, c.settings, lid, group, service)); err != nil {
+				c.log.Error("applying statefulSet", "err", err, "lease", lid, "service", service.Name)
+				return err
+			}
+		} else {
+			if err := applyDeployment(ctx, c.kc, builder.NewDeployment(c.log, c.settings, lid, group, service)); err != nil {
+				c.log.Error("applying deployment", "err", err, "lease", lid, "service", service.Name)
+				return err
+			}
 		}
 
 		if len(service.Expose) == 0 {
 			c.log.Debug("no services", "lease", lid, "service", service.Name)
 			continue
 		}
 
-		serviceBuilderLocal := newServiceBuilder(c.log, c.settings, lid, group, service, false)
-		if serviceBuilderLocal.any() {
+		serviceBuilderLocal := builder.NewService(c.log, c.settings, lid, group, service, false)
```

Comment:
The name of this function sort of conflicts with some of our other code. Normally `NewService` creates an instance of something with an associated goroutine somewhere in our codebase. This `NewService` is actually just a helper for defining kubernetes stuff it appears. I'd prefer it if we could change the name to something less ambiguous.

Reply:
💯